### ISSUE TITLE
Azure Firewall Datasource does not support returning all attributes from equivalent HTTP Get request

### ISSUE THREAD

**Issue by thebenwaters** [hashicorp/terraform-provider-azurerm#16191]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_firewall`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```
N/A 

### Debug Output

N/A 

### Panic Output

N/A 

### Expected Behaviour

Attributes should match https://docs.microsoft.com/en-us/rest/api/firewall/azure-firewalls/get#azurefirewall

### Actual Behaviour

Attributes are listed here https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/firewall

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

This is to track publicly. Separate bug opened internally to track

### References

* #0000

**Comment by magodo** [comment]
@thebenwaters Thank you for submitting this!

We do know there is a gap in data sources comparing to their API model in its OpenAPI spec. In fact, even for managed resources, we are catching up missing features. In most cases, we might more focus on the managed resources in the short term. Whilst in the long term, we are working on some new mechanism, which makes it easier to fill in the feature gaps.

Do you have any specific requirements on the `azurerm_firewall` data source, e.g. which missing attributes you want have access to? So that I can try to make a PR for those.

**Comment by thebenwaters** [comment]
@magodo yes! The most beneficial and crucial requirements for azurerm_firewall data source is the ability to receive policies from within the firewall itself (without the need for the firewall_policy datasource). Please see the [AzureFirewall resource](https://docs.microsoft.com/en-us/rest/api/firewall/azure-firewalls/get#azurefirewall) in the response from the GET call. This gives the customer the ability to [migrate from Azure Firewall Classic to Azure Firewall Standard and Premium](https://docs.microsoft.com/en-us/azure/firewall/premium-migrate#migrate-classic-rules-to-standard-policy)
